File: common/java/src/main/java/org/aiddl/common/learning/LearningTerm.java
package org.aiddl.common.learning;

import org.aiddl.core.representation.SymbolicTerm;
import org.aiddl.core.representation.Term;

public class LearningTerm {
	public static final SymbolicTerm Learner 			= Term.sym("learner");
	public static final SymbolicTerm Predictor			= Term.sym("predictor");
	public static final SymbolicTerm Classifier			= Term.sym("classifier");
	
	public static final SymbolicTerm Data 				= Term.sym("data");
	public static final SymbolicTerm Attributes 		= Term.sym("attributes");
	public static final SymbolicTerm Instance 			= Term.sym("instance");
	public static final SymbolicTerm Label 				= Term.sym("label");

	public static final SymbolicTerm ConfusionMatrix 	= Term.sym("confusion-matrix");
	public static final SymbolicTerm Matrix				= Term.sym("matrix");
	
	public static final SymbolicTerm Fold 				= Term.sym("fold");
	
	public static final SymbolicTerm TrainData 			= Term.sym("train-data");
	public static final SymbolicTerm TestData 			= Term.sym("test-data");
}

File: common/scala/src/main/scala/org/aiddl/common/scala/learning/LearningTerm.scala
package org.aiddl.common.scala.learning

import org.aiddl.core.scala.representation.Sym

enum Term(val sym: Sym):
	case Fit extends Term(Sym("fit"))
	case Predict extends Term(Sym("predict"))
	case SetParams extends Term(Sym("set-params"))



object LearningTerm {
	val Learner 			= Sym("learner")
	val Predictor			= Sym("predictor")
	val Classifier			= Sym("classifier")
	val Class			= Sym("class")
	
	val Data 				= Sym("data")
	val Attributes 		    = Sym("attributes")
	val Instance 			= Sym("instance")
	val Label 				= Sym("label")

	val ConfusionMatrix 	= Sym("confusion-matrix")
	val Matrix				= Sym("matrix")
	
	val Fold 				= Sym("fold")
	
	val TrainData 			= Sym("train-data")
	val TestData 			= Sym("test-data")
}